src/main/java/com/pingcap/tikv/statistics/TiHistogram.java

```diff
@@ -0,0 +1,206 @@
+
+  //Histogram
+  public Histogram histogram = new Histogram();
+
+  //Bucket
+  public static Bucket bucket = new Bucket();
+
+  //createHistogram is func of create Histogram information
+  public static Histogram createHistogram(
+    long tableID, long isIndex, long colID, Snapshot snapshot,TiConfiguration conf,TiCluster cluster) throws Exception{
+    Catalog cat = cluster.getCatalog();
+    TiDBInfo db = cat.getDatabase(DB_NAME);
+    TiTableInfo table = cat.getTable(db, TABLE_NAME);
+    TiIndexInfo index = TiIndexInfo.generateFakePrimaryKeyIndex(table);
+
+    List<TiExpr> firstAnd =
+        ImmutableList.of(
+            new Equal(TiColumnRef.create(TABLE_ID, table), TiConstant.create(tableID)),
+            new Equal(TiColumnRef.create(IS_INDEX, table), TiConstant.create(isIndex)),
+            new Equal(TiColumnRef.create(HIST_ID, table), TiConstant.create(colID)));
+    ScanBuilder scanBuilder = new ScanBuilder();
+    ScanBuilder.ScanPlan scanPlan = scanBuilder.buildScan(firstAnd, index, table);
+    TiSelectRequest selReq = new TiSelectRequest();
+    selReq
+        .addRanges(scanPlan.getKeyRanges())
+        .setTableInfo(table)
+        .addField(TiColumnRef.create(TABLE_ID, table))
+        .addField(TiColumnRef.create(IS_INDEX, table))
+        .addField(TiColumnRef.create(HIST_ID, table))
+        .addField(TiColumnRef.create(BUCKET_ID, table))
+        .addField(TiColumnRef.create(COUNT, table))
+        .addField(TiColumnRef.create(REPEATS, table))
+        .addField(TiColumnRef.create(LOWER_BOUND, table))
+        .addField(TiColumnRef.create(UPPER_BOUND, table))
+        .setStartTs(snapshot.getVersion());
+
+    selReq.addWhere(PredicateUtils.mergeCNFExpressions(scanPlan.getFilters()));
+
+    List<RangeSplitter.RegionTask> keyWithRegionTasks =
+        RangeSplitter.newSplitter(cluster.getRegionManager())
+            .splitRangeByRegion(selReq.getRanges());
+    for (RangeSplitter.RegionTask worker : keyWithRegionTasks) {
+      Iterator<Row> it = snapshot.select(selReq, worker);
+
+      while (it.hasNext()) {
+        SchemaInfer schemaInfer = SchemaInfer.create(selReq);
+        Row row = it.next();
+        long buckID = row.getLong(0);
+        long count = row.getLong(1);
+        long repeats = row.getLong(2);
+        if (isIndex == 1) {
+          bucket.lowerBound = Comparables.wrap(row.getLong(3));
+          bucket.upperBound = Comparables.wrap(row.getLong(4));
+        } else {
+          bucket.lowerBound =
+              (Comparable<ByteString>) row.get(3, DataTypeFactory.of(Types.TYPE_BLOB));
+          bucket.upperBound =
+              (Comparable<ByteString>) row.get(4, DataTypeFactory.of(Types.TYPE_BLOB));
+        }
+      }
+    }
+    cluster.close();
+
+    return new Histogram();
+  }
+
+  // equalRowCount estimates the row count where the column equals to value.
+  public float equalRowCount(ByteString values) {
```

Comment:
Passing values as ByteString is likely too "raw". We don't have a datum (value plus type together) and uses Object directly. So here we should pass Object instead of ByteString.